src/pages/cart/cart.view.ts

```diff
@@ -91,21 +101,68 @@ export class Cart {
                     span.innerText = page.toString();
                     span.setAttribute('data-page', page.toString());
                     this.clearProductsRows();
-                    this.renderProductsRows(productsCart);
+                    this.renderProductsRows();
                 }
             }
         });
+
+        const productsInRows = <HTMLImageElement>document.getElementById('productsRows');
+        productsInRows.addEventListener('click', (event) => {
+            const target = event.target as HTMLElement;
+            const card = target.closest('div.wrap') as HTMLDivElement;
+            const id = card.getAttribute('data-id');
+            if (id && target.className === 'add_to_cart') {
+                const product = this.catalogController.getItemById(parseInt(id));
+                if (product) {
+                    this.cartController.add(product);
+                    this.header.update();
+                    this.clearProductsRows();
+                    this.renderProductsRows();
+                }
+            } else if (id && target.className === 'drop_from_cart') {
+                this.cartController.delete(parseInt(id));
+                this.header.update();
+                this.clearProductsRows();
+                this.renderProductsRows();
+            } else if (id) {
+                const url = window.location.origin;
+                changePagesUrl(url, 'product-details', id);
+            }
+        });
     }
 
-    public renderProductsRows(productsCart: Product[]): void {
+    public renderProductsRows(): void {
+        const productsCart = this.cartController.getProducts();
+        const arrId: number[] = [];
+        const productsCartWithoutRepeats = [];
+
+        productsCart.forEach((element) => arrId.push(element.id));
+        const arrIdWithoutRepeats = new Set(arrId);
+        arrIdWithoutRepeats.forEach((element) => {
+            const product = this.catalogController.getItemById(element);
+            productsCartWithoutRepeats.push(product);
+        });
+
+        arrId.forEach((element) => {
+            debugger;
+            productsCartWithoutRepeats.forEach((item) => {
+                if (element === item.id && !item.number) {
+                    item.number = 1;
+                } else if (element === item.id && item.number) {
+                    item.number += 1;
+                }
+            });
+        });
+        console.log(productsCartWithoutRepeats);
```

Comment:
У тебя уже есть массив с уникальными айли продуктов, тебе нужно пробежаться по этому массиву и вернуть массив продуктов, с дополнительным полем count (у тебя назывался number) - в котором будет кол-во таких продуктов в корзине
Дл этого мы отфильтровываем по каждому айди продукты из корзины и увеличиваем поле count и price. И возвращаем этот обьект
```suggestion
                const productsWithCountItem: ProductWithCount[] = uniqProductId.map((productId) => {
            const productWithCurrentId = productsCart.filter((element) => element.id === productId);
            return productWithCurrentId.reduce<ProductWithCount>((previousValue, currentValue) => {
                const { count = 0, price = 0 } = previousValue;
                return {
                    ...currentValue,
                    count: count + 1,
                    price: price + currentValue.price,
                };
            }, {} as ProductWithCount);
        });
```

Reply:
done